src/components/fields/date-input/date-input.tsx

```diff
@@ -59,6 +61,32 @@ export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
 				.test("not-past", notPastRule?.["errorMessage"] || ERROR_MESSAGES.DATE.CANNOT_BE_PAST, (value) => {
 					if (!isValidDate(value) || !notPastRule?.["notPast"]) return true;
 					return !LocalDate.parse(value).isBefore(LocalDate.now());
+				})
+				.test("min-date", undefined, (value, context) => {
+					const minDate = DateTimeHelper.toLocalDateOrTime(minDateRule?.["minDate"], dateFormat, "date");
+					if (minDate && LocalDate.parse(value).isBefore(minDate)) {
+						return context.createError({
+							message:
+								minDateRule?.["errorMessage"] ||
+								ERROR_MESSAGES.DATE.MIN_DATE(
+									DateTimeHelper.formatDateTime(minDateRule?.["minDate"], "dd/MM/uuuu", "date")
+								),
+						});
+					}
+					return true;
+				})
+				.test("max-date", undefined, (value, context) => {
+					const maxDate = DateTimeHelper.toLocalDateOrTime(maxDateRule?.["maxDate"], dateFormat, "date");
+					if (maxDate && LocalDate.parse(value).isAfter(maxDate)) {
+						return context.createError({
+							message:
+								maxDateRule?.["errorMessage"] ||
+								ERROR_MESSAGES.DATE.MAX_DATE(
+									DateTimeHelper.formatDateTime(maxDateRule?.["maxDate"], "dd/MM/uuuu", "date")
+								),
+						});
+					}
+					return true;
 				}),
```

Comment:
error message is set from within the test functions because the rule value is dynamic

now that I realise the errors can be set this way, I could have combined the rest of the custom validations in one test function, this may give a slight performance boost, we can refactor this in the future